### PR TITLE
Update dependencies and fix e2e tests

### PR DESCRIPTION
**Pull request by Aljullu**
This PR:

* Bumps webpack from 5.831.1 to 5.94.0 (same as PR https://github.com/woocommerce/storefront/pull/2159).
* Updates the version of Node to be in sync with WooCommerce.
* I included `validate-build.js` to the excluded files from builds.
* Updated several other packages that were necessary. That also involved:
  * Disabling a new ESLint rule: `@typescript-eslint/no-this-alias`.
  * Updating the only e2e test we have.
  * Rebuilding `package-lock.json` from scratch.

This allows us to get the e2e tests job pass again.

### How to test the changes in this Pull Request:

This PR doesn't include any new features, it's simply updating dependencies and internal tools.

1. Run `npm ci && npm run build` to build the ZIP.
2. Install the generated ZIP in a fresh store.
3. Do some smoke testing.
4. Verify everything works as it used to and there are no regressions.

### Changelog

> Update many dev dependencies.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,39 +60,50 @@
 			"STORE_URL": "http://localhost:8802"
 		}
 	},
+	"lorem": "ipsum",
```
Comment (dinhtungdu) [inline package.json]: What is this used for?

File: package.json [hunk 2]
```diff
@@ -70,39 +60,50 @@
...
 		"susy": "2.2.14",
-		"typescript": "5.0.4",
-		"uglify-js": "3.17.4"
+		"typescript": "5.5.4",
+		"uglify-js": "3.19.3"
 	},
 	"engines": {
-		"node": "14.21.1",
+		"node": "^20.11.1",
 		"npm": "9.1.3"
```
Comment (dinhtungdu) [inline package.json]: Let's also pump the npm version too, as v10 is ship with node 20

Comment (dinhtungdu) [reply package.json]: https://nodejs.org/en/about/previous-releases

**Review by dinhtungdu — COMMENTED** [review]
Thanks for working on this PR and I'm sorry for the late reviews! I built and smoked test the zip file locally and the built theme works flawlessly in my test, no console log or PHP error thrown. I left some minor comments, otherwise, this is LGTM.

File: package.json (hunk 1 above)
Comment (Aljullu) [reply package.json]: Ups, a leftover from testing. Removed in 54f1178308c7cb46b8c97e4f27927257e6ca998f.

File: package.json (hunk 2 above)
Comment (Aljullu) [reply package.json]: Updated in ea5e7b88f9743727bd81afc5df05af9ba2785931.

**Comment by Aljullu** [comment]
Thanks for the review, @dinhtungdu! This is ready for another look.

**Review by dinhtungdu — APPROVED** [review]
Thanks for the updates! I smoke-tested the build and it still worked as expected for me.

**Pull request merged**